buildcontext/local.go

```diff
@@ -87,21 +86,24 @@ func (lr *localResolver) resolveLocal(ctx context.Context, gwClient gwclient.Cli
 	bf := buildFileValue.(*buildFile)
 
 	var buildContextFactory llbfactory.Factory
-	if _, isTarget := ref.(domain.Target); isTarget {
-		noImplicitIgnore := bf.ftrs != nil && bf.ftrs.NoImplicitIgnore
-		excludes, err := readExcludes(ref.GetLocalPath(), noImplicitIgnore)
-		if err != nil {
-			return nil, err
+	// guard against auto-complete code's GetTargetArgs() func which passes in a nil gwClient (but doesn't actually invoke buildkit)
+	if gwClient != nil {
+		if _, isTarget := ref.(domain.Target); isTarget {
+			noImplicitIgnore := bf.ftrs != nil && bf.ftrs.NoImplicitIgnore
+			excludes, err := readExcludes(ref.GetLocalPath(), noImplicitIgnore)
+			if err != nil {
+				return nil, err
+			}
+			buildContextFactory = llbfactory.Local(
+				ref.GetLocalPath(),
+				llb.ExcludePatterns(excludes),
+				llb.SessionID(gwClient.BuildOpts().SessionID),
```

Comment:
Apparently, this session ID is a different kind of session ID from the buildkit session ID. I remember Tonis mentioned this over the Docker slack some years ago. Could we just not set this at all?

Reply:
Using the SessionID passed in via the BuildOpts was inspired by how the dockerfile frontend works: https://github.com/moby/buildkit/blob/fad945e9b9ae09dbdd110405e1dd78a2c3cbdc09/frontend/dockerfile/builder/build.go#L186

If it's not set we would use `return ls.snapshotWithAnySession(ctx, g)` under https://github.com/moby/buildkit/blob/b3e8c63a48ad8c015f5631fc1947945b229b3919/source/local/local.go#L92

I'm worried that if we simply use any session -- potentially a session from a completely different client (on a different host). I would need to dig in more to fully understand if such a race is possible.

Reply:
I see - yeah, sounds good then. If the Dockerfile frontend does the same then it should be fine. No idea why Buildkit doesn't set that session ID itself but 🤷‍♂️.

Reply:
there's an `InContext` call that jobs get wrapped in, which creates a group containing **only** the current sessionID,

```
func (j *Job) InContext(ctx context.Context, f func(context.Context, session.Group) error) error {
	return f(progress.WithProgress(ctx, j.pw), session.NewGroup(j.SessionID))
}
```

then when `sm.Any` is called, it gets passed that group, which only contains the single session ID; so we're safe to remove this.

> If the Dockerfile frontend does the same

it turns out that the dockerfile frontend is executed by `buildkitd` and not `buildkitctl`, so it appears that the `llb.SessionID` override might only be needed for cases where the server is creating the llb rather than the client.

tl;dr: I'll remove it :)